applications/serial_lte_modem/src/slm_auto_connect.h

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/* 
+ * This file defines network configuration when CONFIG_SLM_AUTO_CONNECT is enabled.
+ */
+
+/* Network-specific default system mode configured by %XSYSTEMMODE */
+0,        /* LTE support */
+1,        /* NB-IoT support */
+0,        /* GNSS support, also define CONFIG_MODEM_ANTENNA if not Nordic DK */
+0,        /* LTE preference */
+/* Network-specific default PDN configured by +CGDCONT and +CGAUTH */
+true,     /* PDP context definition required or not */
+"Non-IP", /* PDP type: ""IP", "IPV6", "IPV4V6", "Non-IP" */
```

Comment:
This is unlikely to work in our test environments. At least 142_12 (thingy:91) uses the same SIM that I am using locally and the Non-IP is not working. These settings will likely need to be changed to format that is supported as widely as possible, so PDP Context definition false and PDP type to "IP".

There is additional " in IP

Reply:
Latest build fails to bring up the PDP-context with these settings.

Reply:
Well, the Non-IP configuration is the default one due to LwM2M over NIDD needs this.
All other type of auto-connect needs to change the conficuration accordingly, as documented.
I agree to "hide" auto-connect feature, only for advanced use cases and explicitly by external overlay.

Reply:
The external overlay is part of our integration tests. Having it like this will break the PR commit tests and nightly tests. 
Options then are: 
1. Removing the overlay tests, which is a step in a wrong direction.
2. Change the default values in here, which will force the customer to modify these values.
3. Add overwriting of the values to our build scripts for integration testing, which is doable, but will further complicate the build scripts.

I would be leaning on 2 here, if customers need to configure SLM anyway.

Reply:
Just summited a new `slm_auto_connect.h` to challenge the CI monster :)